client/src/components/Messaging/UserStatusInfo.js

```diff
@@ -1,14 +1,29 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import styles from './styles/UserStatusInfo.module.css';
+import { ThemeContext } from '../../ThemeContext.js';
+import { FiSun, FiMoon } from 'react-icons/fi';
 
-export const UserStatusInfo = ({ online }) => (
-  <div className={styles.userInfo}>
-    {online ? (
-      <span className={styles.userInfoOnline}>
-        Alice {'<'}Online{'>'}
+export const UserStatusInfo = ({ online }) => {
+  const [darkMode, setDarkMode] = useContext(ThemeContext);
+  const toggleTheme = () => {
+    setDarkMode(!darkMode);
+  };
+  return (
+    <div className={styles.userInfo}>
+      {online ? (
+        <span className={styles.userInfoOnline}>
+          Alice {'<'}Online{'>'}
+        </span>
+      ) : (
+        'Waiting for Alice to join..'
+      )}
+      <span className={styles.toggleIcon}>
```

Comment:
Create a separate component for the theme toggling button (including the context setter). 
Re-use the button on both pages i.e. link generation and messaging.

Reply:
Will do this.

Reply:
Done